google-cloud-storage/lib/google/cloud/storage/bucket.rb

```diff
@@ -1448,18 +1454,54 @@ def compose sources, destination, acl: nil, encryption_key: nil
         #                                    "x-goog-meta-foo" => "bar,baz"
         #                                  }
         #
-        def signed_url path, method: nil, expires: nil, content_type: nil,
+        # @example Generating a signed URL for resumable upload:
+        #   require "google/cloud/storage"
+        #
+        #   storage = Google::Cloud::Storage.new
+        #
+        #   bucket = storage.bucket "my-todo-app"
+        #   url = bucket.signed_url "avatars/heidi/400x400.png",
+        #                           method: "POST",
+        #                           content_type: "image/png",
+        #                           headers: {
+        #                             "x-goog-resumable" => "start"
+        #                           }
+        #   # Send the `x-goog-resumable:start` header and the content type
+        #   # with the resumable upload POST request.
+        #
+        # @example Omitting `path` for a URL to list all files in the bucket.
+        #   require "google/cloud/storage"
+        #
+        #   storage = Google::Cloud::Storage.new
+        #
+        #   bucket = storage.bucket "my-todo-app"
+        #   list_files_url = bucket.signed_url version: :v4
+        #
+        def signed_url path = nil, method: nil, expires: nil, content_type: nil,
```

Comment:
@blowmage Just want to draw your attention to the change to the method signature here: `path` -> `path = nil`. This enables the syntax for generating a URL to list all files in the bucket: `list_files_url = bucket.signed_url version: :v4`

Reply:
Yes, I saw that discussion. I think this makes total sense.

Reply:
Does this work for both `v2` and `v4`?

Reply:
Yes, see [signed_url_test.rb#L207-L224](https://github.com/googleapis/google-cloud-ruby/blob/9537343f7d4796c0c1e1a6d3976c3a9711f88893/google-cloud-storage/acceptance/storage/signed_url_test.rb#L207-L224).